distributed-commandbus-springcloud/src/main/java/org/axonframework/springcloud/commandhandling/SpringHttpCommandBusConnector.java

```diff
@@ -181,7 +219,75 @@ public void onSuccess(CommandMessage<? extends C> commandMessage, R result) {
         }
-
     }
 
+    /**
+     * Builder class to instantiate a {@link SpringHttpCommandBusConnector}.
+     * The {@code localCommandBus} of type (@link CommandBus}, {@link RestOperations} and {@link Serializer} are
+     * <b>hard requirements</b> and as such should be provided.
+     */
+    public static class Builder {
+
+        private CommandBus localCommandBus;
+        private RestOperations restOperations;
+        private Serializer serializer;
+
+        /**
+         * Sets the {@code localCommandBus} of type {@link CommandBus} to publish received commands which to the local
+         * segment.
+         *
+         * @param localCommandBus the {@link CommandBus} to publish received commands which to the local segment
+         * @return the current Builder instance, for a fluent interfacing
+         */
+        public Builder localCommandBus(CommandBus localCommandBus) {
+            assertNonNull(localCommandBus, "LocalCommandBus may not be null");
+            this.localCommandBus = localCommandBus;
+            return this;
+        }
+
+        /**
+         * Sets the {@link RestOperations} used to send commands to other nodes.
+         *
+         * @param restOperations the {@link RestOperations} used to send commands to other nodes
+         * @return the current Builder instance, for a fluent interfacing
+         */
+        public Builder restOperations(RestOperations restOperations) {
+            assertNonNull(restOperations, "RestOperations may not be null");
+            this.restOperations = restOperations;
+            return this;
+        }
+
+        /**
+         * Sets the {@link Serializer} used to serialize command messages when they are sent between nodes.
+         *
+         * @param serializer the {@link Serializer} used to serialize command messages when they are sent between nodes
+         * @return the current Builder instance, for a fluent interfacing
+         */
+        public Builder serializer(Serializer serializer) {
+            assertNonNull(serializer, "Serializer may not be null");
+            this.serializer = serializer;
+            return this;
+        }
+
+        /**
+         * Initializes a {@link SpringHttpCommandBusConnector} as specified through this Builder.
+         *
+         * @return a {@link SpringHttpCommandBusConnector} as specified through this Builder
+         */
+        public SpringHttpCommandBusConnector build() {
+            return new SpringHttpCommandBusConnector(this);
+        }
+
+        /**
+         * Validate whether the fields contained in this Builder as set accordingly.
+         *
+         * @throws AxonConfigurationException if one field is asserted to be incorrect according to the Builder's
+         *                                    specifications
+         */
+        protected void validate() {
+            assertNonNull(localCommandBus, "The localCommandBus is a hard requirement and should be provided");
```

Comment:
nit: `localCommandBus` starts with a small letter and the RestOperations and Serializer with capitals. 

Should it all be variable names and be `localCommandBus`, `restOperations` and `serializer` or should it be all component based names and be `CommandBus`, `RestOperations` and `Serializer`?

Reply:
Differs per use case.
If I'd state the `CommandBus` isn't set, that would cause confusion, so `local` should be added there. Cut corners on this guy a little though, as I agree that `local CommandBus` makes more sense.

I figured that place where the field's type itself is enough description, like `RestOperations` and `Serializer`, I'd leave the capitalized to point out the actual type which is missing.